peachjam/views/courts.py

```diff
@@ -108,3 +116,58 @@ def get_context_data(self, **kwargs):
 
 class CourtYearView(CourtDetailView):
     queryset = Judgment.objects.prefetch_related("judges")
+
+
+class CourtRegistryDetailView(CourtDetailView):
+    queryset = Judgment.objects.prefetch_related("judges")
+    template_name = "peachjam/court_registry_detail.html"
+
+    def get_base_queryset(self):
+        qs = super().get_base_queryset().filter(registry=self.registry)
+        if "year" in self.kwargs:
+            qs = qs.filter(date__year=self.kwargs["year"])
+        return qs
+
+    def get_queryset(self):
+        self.court = get_object_or_404(Court, code=self.kwargs["code"])
+        self.registry = get_object_or_404(
+            CourtRegistry, code=self.kwargs["registry_code"]
+        )
+        return super().get_queryset()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["registry"] = self.registry
+        context["formatted_registry_name"] = self.registry.name
+        context["court"] = self.court
+
+        years = self.queryset.filter(registry=self.registry).dates(
+            "date", "year", order="DESC"
+        )
+
+        context["years"] = list(
+            {
+                "url": reverse(
+                    "court_registry_year",
+                    args=[self.court.code, self.registry.code, y.year],
+                ),
+                "year": y.year,
+            }
+            for y in years
+        )
+        context["all_years_url"] = reverse(
+            "court_registry", args=[self.court.code, self.registry.code]
+        )
+
+        if "year" in self.kwargs:
+            context["year"] = self.kwargs["year"]
+            context["formatted_registry_name"] = (
```

Comment:
Do this at the same point as when you first build up `formatted_registry_name` above, rather than separately here.